_posts/2021-05-07-running-multinode-modules-gcp.md

```diff
@@ -0,0 +1,174 @@
+
+* Rspec labelling
+  How do we differentiate acceptance tests from integration tests?
+  Using rspec labeling we can label tests, e.g. the test is tagged as integration. Everything else can stay the same.
+
+  ![Example for tagging tests]({% link /assets/2021-05-07-running-multinode-modules-gcp/integration_tag.png %})
+
+* Integration test rake task
+  Adding a friendly rake task.This is added to the Rakefile of the module.
+
+  ![Example for adding rake task]({% link /assets/2021-05-07-running-multinode-modules-gcp/rake_task.png %})
+
+* Helper functions to target labeled containers/vms
+  There are a number of helper methods required to allow tests to target a specific container/vm.
+  This is required for serverspec/litmus as it will be carrying out the tests on that container/vm.
+  Below are some examples or helper methods of filtering to get either a single vm/container or retrieving multiple vms/targets.
+  [puppetlabs-websphere_application_server](https://github.com/puppetlabs/puppetlabs-websphere_application_server/blob/main/spec/spec_helper_acceptance_local.rb)
+  [puppetlabs-kubernetes](https://github.com/puppetlabs/puppetlabs-kubernetes/blob/main/spec/spec_helper_acceptance_local.rb)
+
+  ![Example for identifying target nodes]({% link /assets/2021-05-07-running-multinode-modules-gcp/target_nodes.png %})
+
+* Testing
+   The below command will only run tests labelled as ‘integration’
+   `bundle exec rake websphere_application_server:integration`
+
+***Teardown***
+
+  We use existing provision task to teardown all the provisioned machines.
+
+## Lets go through different scenarios where we can use multi node testing
+
+***Multilpe puppet agents***
+
+In the setup we have many puppet agents, a module is installed and tests are run.
+
+Commands
+```
+bundle install --path .bundle/gems/ --jobs 4
+bundle exec rake spec_prep
+bundle exec bolt --modulepath spec/fixtures/modules plan run ntp::provision_gcp
+bundle exec rake litmus:install_agent
+bundle exec rake litmus:install_module
+bundle exec rake ntp::integration
+bundle exec rake 'litmus:tear_down'
+```
+Example
+[puppetlabs-ntp](https://github.com/puppetlabs/puppetlabs-ntp/tree/multinodentp)
+
+***Puppet server and agent***
+
+In the setup we have puppet server and puppet agents, a module is installed and tests are run.
+we can use bolt plans for provisioning number of nodes with roles tagged to each of it.
+There is new task to install open source puppet server in the provision module.
+[install_puppetserver](https://github.com/puppetlabs/provision/blob/main/tasks/install_puppetserver.json)
+
+Commands
+```
+bundle install --path .bundle/gems/ --jobs 4
+bundle exec rake spec_prep
+bundle exec bolt --modulepath spec/fixtures/modules plan run kubernetes::provision_cluster
+bundle exec bolt --modulepath spec/fixtures/modules plan run kubernetes::puppetserver_setup
```

Comment:
don't these also need `--inventory spec/fixtures/litmus_inventory.yaml` ?

Reply:
absolutely. Adding the changes . Thank you